### PR TITLE
DOCS-9947-http-work-threading-profile-kt

### PR DESCRIPTION
**Pull request by kevintroller**
Adding worker-threading-profile configurable attributes for HTTP Listener (Mule 3)

### CODE REVIEW

**Review by JuLuzzi — DISMISSED** [review]

File: http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc [hunk 1]
```diff
@@ -1364,6 +1364,50 @@ image::http-listener-connector-51981.png[]
 
 Although this default configuration is usually enough, it is possible to tune the max thread value ^2^
 
+=== Worker Threading Profile
+
+Attributes of Worker Threading Profile (`worker-threading-profile`) parameter:
```
Comment (dhirsch1) [inline http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc]: the Worker Threading Profile...

File: http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc [hunk 2]
```diff
@@ -1364,6 +1364,50 @@ image::http-listener-connector-51981.png[]
...
+|Max Active Thread | Integer |The maximum number of threads to use. | No |16
+
+|Max Idle Threads | Integer |The maximum number of idle or inactive threads that can be in the pool before they are destroyed. | No | 1
+
+|Thread TTL | Long |Determines how long an inactive thread is kept in the pool before being discarded. | No | 60000
+
+|Pool Exhausted Action | WAIT, DISCARD, DISCARD_OLDEST, ABORT, RUN | When the maximum pool size or queue size is bounded, this value determines how to manage incoming tasks. Possible values are:
+
+* `WAIT` +
+Wait until a thread becomes available; don't use this value if the minimum number of threads is zero, in which case a thread may never become available.
```
Comment (dhirsch1) [inline http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc]: The ";" makes a run-on sentence. I would split into two sentences.

File: http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc [hunk 3]
```diff
@@ -1364,6 +1364,50 @@ image::http-listener-connector-51981.png[]
...
+
+* `DISCARD` +
+Throw away the current request and return.
```
Comment (dhirsch1) [inline http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc]: No period when not a sentence - global

File: http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc [hunk 4]
```diff
@@ -1364,6 +1364,50 @@ image::http-listener-connector-51981.png[]
...
+* `DISCARD_OLDEST` +
+Throw away the oldest request and return.
+
+* `ABORT` +
+ Throw a RuntimeException.
+
+* `RUN` +
+The thread making the execute request runs the task itself, which helps guard against lockup. | No | `RUN`
+
+|Thread Wait Timeout | Long |How long to wait in milliseconds when the pool exhausted action is WAIT. If the value is negative, it waits indefinitely. | No | 30000
```
Comment (dhirsch1) [inline http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc]: Wait for what?

File: http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc [hunk 5]
```diff
@@ -1364,6 +1364,50 @@ image::http-listener-connector-51981.png[]
...
+
+|Max Buffer Size | Integer |Determines how many requests to queue when the pool is at maximum usage capacity and the pool exhausted action is WAIT. The buffer throttles for thread creation, before requests are processed.  <<mbsinfo,Do not use maxBufferSize without reading these warnings>>.
```
Comment (dhirsch1) [inline http/0.3.9/modules/ROOT/pages/http-listener-connector.adoc]: Remove the comma in the second sentence.

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**